doc/src/documentation_style_guide.rst

```diff
@@ -0,0 +1,1019 @@
+>>> from sympy.abc import x
+>>> f = Function('f')
+>>> dsolve(cos(f(x)) - (x*sin(f(x)) - f(x)**2)*f(x).diff(x),
+... f(x), hint='1st_exact')
+Eq(x*cos(f(x)) + f(x)**3/3, C1)
+
+Here ``dsolve(cos(f(x)) - (x*sin(f(x)) - f(x)**2)*f(x).diff(x), f(x), hint='1st_exact')`` is too long, so we line break it after a comma so that it
+is readable, and put ``...`` on the continuation lines. If this is not done
+correctly, the doctests will fail.
+
+The output of a command can also be line wrapped. No ``...`` should be used in
+this case. The doctester automatically accepts output that is line wrapped.
+
+**Example**
+
+>>> list(range(30))
+[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21,
+22, 23, 24, 25, 26, 27, 28, 29]
+
+In general, you should run ``./bin/doctest`` to make sure your examples run
+correctly, and fix them if they do not.
+
+4. Parameters Section
+---------------------
+
+This section is encouraged. If you choose to include a Parameters section in
+your docstring, it should be labeled with the heading “Parameters” underlined
+with the same length in equals signs.
+
+::
+
+  Parameters
+  ==========
+
+If you have parameters listed in parentheses after a function, class, or method
+name, you must include a parameters section.
+
+This section consists of descriptions of the function arguments, keywords, and
+their respective types.
+
+Enclose variables in single backticks. The colon must be preceded by a space,
+or omitted if the type is absent. For the parameter types, be as precise as
+possible. If it is not necessary to specify a keyword argument, use
+``optional``. Optional keyword parameters have default values, which are
+displayed as part of the function signature. They can also be detailed in the
+description.
+
+When a parameter can only assume one of a fixed set of values, those values can
+be listed in braces, with the default appearing first. When two or more input
+parameters have exactly the same type, shape, and description, they can be
+combined.
+
+If the Parameters section is not formatted correctly, the documentation build
+will render incorrectly.
+
+WORK IN PROGRESS via issue `#17644
+<https://github.com/sympy/sympy/issues/17644>`_ and via related issue `#17618
+<https://github.com/sympy/sympy/issues/17618>`_: Returns can be listed as a
+subheading within the Parameters section.
+
+**Example**
+
+Here is an example of a correctly formatted Parameters section::
+
+  def opt_cse(exprs, order='canonical'):
+      """
+      Find optimization opportunities in Adds, Muls, Pows and negative
+      coefficient Muls.
+
+      Parameters
+      ==========
+
+      exprs : list of sympy expressions
+          The expressions to optimize.
+      order : string, 'none' or 'canonical'
+          The order by which Mul and Add arguments are processed. For large
+          expressions where speed is a concern, use the setting order='none'.
+
+      """
+
+5. See Also Section
+-------------------
+
```

Comment:
I noticed when fixing https://github.com/sympy/sympy/pull/17698 that sometimes people mix up "see also" and "references". See also can only be used to reference other SymPy objects. Anything that is a link has to go in "references". See for instance in sympy/geometry/parabola.py (which is currently not included in the Sphinx docs).

Reply:
I'll add a note about this!